ios/MaasNearbyTransit/Sources/MaasNearbyTransit/Components/NearbyTransitStopBadgeView.swift

```diff
@@ -0,0 +1,99 @@
+import SwiftUI
+import Swappable
+
+struct NearbyTransitStopBadgeView: View, Swappable {
+
+    var constants: Kotlin<NearbyTransitStopBadgeConstants> { Kotlin(NearbyTransitStopBadgeConstants()) }
+
+    public struct InputType {
+        public let schedules: [Schedule]
+    }
+    let input: InputType
+
+    public init(input: InputType) {
+        self.input = input
+    }
+
+    public init(schedules: [Schedule]) {
+        self.input = .init(schedules: schedules)
+    }
+
+    private var slicedFilteredSchedules: [Schedule] {
+        input.schedules.filterDuplicates(includeElement: { $0.transportId == $1.transportId })
+    }
+
+    var defaultBody: some View {
+        ZStack {
+            ForEach(slicedFilteredSchedules.indices) {
+                imageView(schedule: slicedFilteredSchedules[$0])
+                    .padding([.leading, .bottom], $0 != 0 ? CGFloat((-constants.offsetXY * CGFloat($0))) : 0)
+            }
+        }
+    }
+
+    private func imageView(schedule: Schedule) -> some View {
+        Image.from(image: UIImage(named: schedule.transportId))
+            .frame(width: constants.imageWidth, height: constants.imageHeight, alignment: .center)
+            .background(schedule.color.parseColor())
+            .foregroundColor(Color.white)
+    }
+}
+
+private extension Array {
+
+    func filterDuplicates(includeElement: (_ lhs:Element, _ rhs:Element) -> Bool) -> [Element] {
```

Comment:
In other languages this is usually called `distinct` / `distinctBy`, a bit surprised Swift doesn't have this out of the box 😮

Reply:
There is `distinct` when `Element is Comparable` 🤔

Reply:
Can we make the Obj-C `Schedule` conform to `Comparable`?